### ISSUE TITLE
Cachalot attempting to cache databases which are not supported and not present in CACHALOT_DATABASES

### ISSUE THREAD

**Issue by jjorissen52** [noripyt/django-cachalot#106]
Cachalot is attempting to cache databases which are not supported and not present in CACHALOT_DATABASES. I have to use `django-sqlserver` to connect to my company's database. I am sure that it is unsupported, and that is completely understandable. However, I am running into an issue where `cachalot` attempts to cache queries for the unsupported database, regardless of the content of my `CACHALOT_DATABASES` setting.

Any help would be greatly appreciated!

## settings.py
```
DATABASES =[
    . . .
    'mssql_db': {
            'ENGINE': 'sqlserver',
            'HOST': config.get(DB_REGION, 'ms_host'),
            'NAME': MS_DB,
            'USER': config.get(DB_REGION, 'ms_user'),
            'PASSWORD': config.get(DB_REGION, 'ms_password'),
    },
]
CACHALOT_DATABASES = ['default', ]
```

## models.py
```
class DeficientChart(models.Model):

    . . . 

    class Meta:
        managed = True if settings.RUNNING_UNIT_TESTS else False
        db_table = '[tmp].[DeficientChart]'

```

## Test Code
```
current_chart_dict = current_chart.values()[0]
current_chart_dict['tmpStatus'] = update_list_data[0]['tmpStatus']
processed_chart_dict = current_chart.values()[0]
processed_chart_dict['Status'] = update_list_data[0]['tmpStatus']
# DeficientChart constructor used so the model __init__ method will do validation for us
# since django-sqlserver does not seem to support transactions...........
validated_current_chart = DeficientChart(**current_chart_dict)
new_processed_chart = ProcessedChart(**processed_chart_dict)

print(settings.CACHALOT_DATABASES)
current_chart.update(tmpStatus=update_list_data[0]['tmpStatus'])
new_processed_chart.save(force_insert=True)
```

## Output/traceback
```
['default']

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/Projects/my_app/venv/lib/python3.6/site-packages/django/apps/registry.py in get_app_config(self, app_label)
    148         try:
--> 149             return self.app_configs[app_label]
    150         except KeyError:

KeyError: '[tmp]'

During handling of the above exception, another exception occurred:

LookupError                               Traceback (most recent call last)
<ipython-input-5-cf8141da2b0b> in <module>()
     17 
     18 print(settings.CACHALOT_DATABASES)
---> 19 current_chart.update(tmpStatus=update_list_data[0]['tmpStatus'])
     20 new_processed_chart.save(force_insert=True)

~/Projects/my_app/venv/lib/python3.6/site-packages/django/db/models/query.py in update(self, **kwargs)
    645         query._annotations = None
    646         with transaction.atomic(using=self.db, savepoint=False):
--> 647             rows = query.get_compiler(self.db).execute_sql(CURSOR)
    648         self._result_cache = None
    649         return rows

~/Projects/my_app/venv/lib/python3.6/site-packages/cachalot/monkey_patch.py in inner(compiler, *args, **kwargs)
     30         compiler.connection.raw = False
     31         try:
---> 32             return original(compiler, *args, **kwargs)
     33         finally:
     34             compiler.connection.raw = True

~/Projects/my_app/venv/lib/python3.6/site-packages/cachalot/monkey_patch.py in inner(write_compiler, *args, **kwargs)
     92         if is_cachable(table):
     93             invalidate(table, db_alias=db_alias,
---> 94                        cache_alias=cachalot_settings.CACHALOT_CACHE)
     95         return original(write_compiler, *args, **kwargs)
     96 

~/Projects/my_app/venv/lib/python3.6/site-packages/cachalot/api.py in invalidate(*tables_or_models, **kwargs)
     69     invalidated = set()
     70     for cache_alias, db_alias, tables in _cache_db_tables_iterator(
---> 71             _get_tables(tables_or_models), cache_alias, db_alias):
     72         cache = cachalot_caches.get_cache(cache_alias, db_alias)
     73         if not isinstance(cache, AtomicCache):

~/Projects/my_app/venv/lib/python3.6/site-packages/cachalot/api.py in _get_tables(tables_or_models)
     33     return [(apps.get_model(o)._meta.db_table if '.' in o else o)
     34             if isinstance(o, string_types) else o._meta.db_table
---> 35             for o in tables_or_models]
     36 
     37 

~/Projects/my_app/venv/lib/python3.6/site-packages/cachalot/api.py in <listcomp>(.0)
     33     return [(apps.get_model(o)._meta.db_table if '.' in o else o)
     34             if isinstance(o, string_types) else o._meta.db_table
---> 35             for o in tables_or_models]
     36 
     37 

~/Projects/my_app/venv/lib/python3.6/site-packages/django/apps/registry.py in get_model(self, app_label, model_name, require_ready)
    198             app_label, model_name = app_label.split('.')
    199 
--> 200         app_config = self.get_app_config(app_label)
    201 
    202         if not require_ready and app_config.models is None:

~/Projects/my_app/venv/lib/python3.6/site-packages/django/apps/registry.py in get_app_config(self, app_label)
    154                     message += " Did you mean '%s'?" % app_config.label
    155                     break
--> 156             raise LookupError(message)
    157 
    158     # This method is performance-critical at least for Django's test suite.

LookupError: No installed app with label '[tmp]'.
```

**Comment by BertrandBordage** [comment]
That's interesting!
Your description is not entirely accurate, django-cachalot is not trying to cache Microsoft SQL Server queries, it's only trying to invalidate them. That's done on purpose, this is the expected behaviour.

However, there is something I didn't anticipate: I didn't know explicit Django Model Meta table names could have a `.` in it.

I'll write a test and figure out a solution. In the meantime, if possible, change your table name in Meta so it doesn't contain a dot. Or simply wait before integrating django-cachalot.

**Comment by BertrandBordage** [comment]
Fixed by 66f6e2003c5104b5aa8c2e564f1aeeb7abb2378e.

**Comment by jjorissen52** [comment]
Wow that was fast. I will get the changes and test it out ASAP.

**Comment by BertrandBordage** [comment]
Fix available in the new 2.0.1 release!
